### ISSUE TITLE
Improper "unquoted node name found"

### ISSUE THREAD

**Issue by cyberious** [rodjek/puppet-lint#904]
I have a function that is used for ExecutionResult sets and checks them... and I am getting unquoted node name found b/c the function is written in puppet.

Exepectation is to either ignore these in a function or fix the issue with regards to this in a function.

functions/process_result.pp:57:38: unquoted node name found

```
// offending declaration is reported as the `.ok` 
if ($abort_on_errors and !$results.ok) { 
    alert(@(EOT))
      Aborting plan due to task failure
    | EOT
    pe::exit(1)
  }```

**Comment by usev6** [comment]
I'm not able to reproduce this problem with the code you pasted (neither on HEAD nor with one of the recent releases).

```
$ cat gh/manifests/issue904.pp
## gh issue 904
class gh::issue904 {
  if ($abort_on_errors and !$results.ok) {
    alert(@(EOT))
      Aborting plan due to task failure
    | EOT
    pe::exit(1)
  }
}
$ puppet-lint gh/manifests/issue904.pp 
WARNING: top-scope variable being used without an explicit namespace on line 3
WARNING: top-scope variable being used without an explicit namespace on line 3
```

@cyberious could you maybe provide a more complete example that shows the failure mode? Also, what version of puppet-lint do you use?

**Comment by cyberious** [comment]
When the autofix is enabled it attempts to do `!$results.'ok'` sorry I wasn't clear earlier

**Comment by rodjek** [comment]
Closing for now, please feel free to reopen again with more information.
